pysrc/bytewax/run.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import importlib
+import pathlib
+import os
+import sys
+
+from bytewax.recovery import SqliteRecoveryConfig
+
+from .bytewax import cluster_main, spawn_cluster
+
+__all__ = ["cluster_main"]
+
+
+class ImportFromStringError(Exception):
+    pass
+
+
+def import_from_string(import_str):
+    if not isinstance(import_str, str):
+        return import_str
+
+    help_msg = (
+        f"Import string '{import_str}' must be in format "
+        "'<module>:<attribute>' or '<module>:<factory function>:<optional string arg>."
+    )
+
+    if import_str.count(":") > 2:
+        raise ImportFromStringError(help_msg)
+
+    module_str, _, attrs_str = import_str.partition(":")
+    if not module_str or not attrs_str:
+        raise ImportFromStringError(help_msg)
+
+    try:
+        module = importlib.import_module(module_str)
+    except ImportError as exc:
+        if exc.name != module_str:
+            raise exc from None
+        raise ImportFromStringError(f'Could not import module "{module_str}".')
+
+    instance = module
+    try:
+        for attr_str in attrs_str.split("."):
+            if ":" in attr_str:
+                attr_str, _, arg = attr_str.partition(":")
+                func = getattr(instance, attr_str)
+                if not callable(func):
+                    raise ImportFromStringError(
+                        f"Factory function `{attr_str}` is not a function"
+                    ) from None
+                instance = func(arg)
+            else:
+                instance = getattr(instance, attr_str)
+    except AttributeError:
+        raise ImportFromStringError(
+            f"Attribute '{attrs_str}' not found in module '{module_str}'."
+        ) from None
+
+    return instance
+
+
+class EnvDefault(argparse.Action):
+    """Action that uses env variable as default if nothing else was set."""
+
+    def __init__(self, envvar, default=None, **kwargs):
+        if envvar:
+            default = os.environ.get(envvar, default)
+        super(EnvDefault, self).__init__(default=default, **kwargs)
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, values)
+
+
+def _prepare_import(import_str):
```

Comment:
This too was added after Esteban's review to avoid breaking the current helm chart. I don't like it too much, but I took the method from flask's codebase so that at least it behaves in a familiar way for python people, and it seems to work correctly even though I agree it looks fragile and may break in some cases.